sub/sd_ass.c

```diff
@@ -272,9 +278,30 @@ static void decode(struct sd *sd, struct demux_packet *packet)
     } else {
         // Note that for this packet format, libass has an internal mechanism
         // for discarding duplicate (already seen) packets.
-        ass_process_chunk(track, packet->buffer, packet->len,
-                          llrint(packet->pts * 1000),
-                          llrint(packet->duration * 1000));
+        if (sd->opts->sub_filter_SDH) {
+            if (packet->len + 1 > ctx->fbuffer_len) {
+                free(ctx->fbuffer);
+                ctx->fbuffer_len = (((packet->len + 1) >> 8) + 1) << 8;
+                ctx->fbuffer = malloc(ctx->fbuffer_len);
+            }
+            if (!ctx->fbuffer) {
+                ass_process_chunk(track, packet->buffer, packet->len,
+                                  llrint(packet->pts * 1000),
+                                  llrint(packet->duration * 1000));
+            } else {
+                memcpy(ctx->fbuffer, packet->buffer, packet->len);
+                ctx->fbuffer[packet->len] = '\0';
```

Comment:
My filter need a null terminated string and I wanted to avoid allocating memory, calling filter and then discarding memory for every subtitle. So the buffer was for having data that could be reused for all subtitle lines. But I think that in mpv it is very common to allocate mamory, copy data there and the discard it again instead of trying to avoid copying/allocating.